src/gt4py/next/program_processors/runners/dace_fieldview/transformations/strides.py

```diff
@@ -426,59 +435,137 @@ def next_edges_by_connector(
+    #  NestedSDFG. However, it could be that the strides is used somewhere else.
+    # We will do the following, if `ignore_symbol_mapping` is `False` and
+    #  the strides of the inner descriptors are symbols, we will use the
+    #  symbol mapping. Otherwise, we will replace the `strides` attribute
+    #  of the inner descriptor, in addition we will install a remapping,
+    #  for those values that were a symbol.
     if (not ignore_symbol_mapping) and all(
-        isinstance(inner_stride, dace.symbol) for inner_stride in inner_desc.strides
+        isinstance(inner_stride, dace.symbol) for inner_stride in inner_strides_init
     ):
+        # Use the symbol
         for inner_stride, outer_stride in zip(inner_desc.strides, new_strides, strict=True):
             nsdfg_node.symbol_mapping[inner_stride.name] = outer_stride
     else:
-        assert len(inner_desc.shape) == len(new_strides)
+        # We have to replace the `strides` attribute of the inner descriptor.
         inner_desc.set_shape(inner_desc.shape, new_strides)
 
-        new_strides_symbols: list[dace.symbol] = functools.reduce(
-            lambda acc, itm: (acc + list(itm.free_symbols))  # type: ignore[union-attr]
-            if dace.symbolic.issymbolic(itm)
-            else acc,
-            new_strides,
-            [],
-        )
-        new_strides_free_symbols = {
-            sym for sym in new_strides_symbols if sym.name not in nsdfg_node.sdfg.symbols
+        # Now find the free symbols that the new strides need.
+        new_strides_symbols: list[str] = []
+        for new_stride_dim in new_strides:
+            if dace.symbolic.issymbolic(new_stride_dim):
+                new_strides_symbols.append(str(new_stride_dim))
+            else:
+                new_strides_symbols.extend(sym for sym in new_stride_dim.free_symbols)
+
+        # Now we determine the set of symbols that should be mapped inside the NestedSDFG.
+        #  We will exclude all that are already inside the `symbol_mapping` (we do not
+        #  check if they map to the same value, we just hope it). Furthermore,
+        #  we will exclude all symbols that are listed in the `symbols` property
+        #  of the SDFG that is nested, and hope that it has the same meaning.
+        missing_symbol_mappings: set[str] = {
+            sym
+            for sym in new_strides_symbols
+            if not (sym in nsdfg_node.sdfg.symbols or sym in nsdfg_node.symbol_mapping)
         }
-        for sym in new_strides_free_symbols:
-            nsdfg_node.sdfg.add_symbol(sym.name, sym.dtype)
-            nsdfg_node.symbol_mapping[sym.name] = sym
+        for sym in missing_symbol_mappings:
+            # We can not create symbols in the nested SDFG, because we do not have
+            #  the type of the symbols.
+            nsdfg_node.symbol_mapping[sym] = dace.symbolic.pystr_to_symbolic(sym)
+
+        # Now create aliases for the old symbols that were used as strides.
```

Comment:
It should be safe to remove them. There is a `used_symbols()` function, if I remember right. So we could assert that they are not used.

Reply:
I think I will keep it.
Using `used_symbols()` on the SDFG, will be very expensive as the whole thing has to be scanned.